lib/index.js

```diff
@@ -45,6 +45,26 @@ var merge = curryN(2, function(s1, s2) {
   return s;
 });
 
+function mergeAll(streams) {
+  var s = immediate(combine(function() {
+    var self = arguments[arguments.length - 2];
+    if (arguments[arguments.length - 1][0]) {
+      self(arguments[arguments.length - 1][0]());
+      return;
+    }
+    [].slice.call(arguments, streams.length).some(function(s1) {
+      if (s1.hasVal) {
+        self(s1.val);
+        return true;
+      }
+    });
+  }, streams));
+  endsOn(combine(function() {
+    return true;
+  }, streams.map(function(sm) { return sm.end; })), s);
```

Comment:
Isn't that what is happening here? I've tested locally and I believe this is the same logic used for `merge`.